### ISSUE TITLE
Audio not working on iOS unless bluetooth headphones are connected

### ISSUE THREAD

**Issue by andyvannoy** [YoYoGames/GameMaker-Bugs#4943]
### Description

It doesn't matter if I play the HTML5 version of my game, or export it to iOS. Either way I can't hear any audio at all. 
I have tried setting the priority to 100: 
audio_play_sound(global.target_play, 100, false);

This is not autoplay audio, audio is directly played from buttons.
My audio files are mp3's, set as uncompress - not streamed.

### Steps To Reproduce

Try playing my game on pc vs safari on iphone:

https://www.dougpeaches.com/games

All I did was make a game using the audio_play_sound function

contact us package:
[package.zip](https://github.com/YoYoGames/GameMaker-Bugs/files/14532726/package.zip)

### How reliably can you recreate this issue using your steps above?

Always

### Which version of GameMaker are you reporting this issue for?

2024.2.0 (Monthly)

### Which platform(s) are you seeing the problem on?

iOS

### Contact Us Package Attached?

- [X] I have attached my Contact Us Package

### Sample Project Added?

- [ ] I have included a small sample project

**Comment by kadinshino** [comment]
I belive this is related to needing to access  AVAudioSession.sharedInstance() for iOS within Gamemaker. I'm not sure if anyone knows of an extension that manages this.

**Comment by SamTHodson** [comment]
I was unable to reproduce this issue with the linked game and with my own projects on an iPhone 8 and a 15.

@andyvannoy would you be able to confirm what device you are using and ensure that it's not set to any mode which might interfere with audio?

**Comment by andyvannoy** [comment]
> I was unable to reproduce this issue with the linked game and with my own projects on an iPhone 8 and a 15.
> 
> @andyvannoy would you be able to confirm what device you are using and ensure that it's not set to any mode which might interfere with audio?

yeah, it was the silent mode switch. do you know if its possible to make audio play with the silent mode switch flipped on ios? plenty of games do this? thanks for the reply

**Comment by stuckie** [comment]
We should add the option for handling silent mode.

**Comment by andyvannoy** [comment]
> We should add the option for handling silent mode.

Hey that would be awesome.

Just to add some more info, silent mode blocks sound in both native iOS exports and in html5 websites in safari. Would be great to have a toggle to change both of those. 

Thanks much for looking into this. I want to make a bunch of sound-based music theory games and fixing this would help my users.

**Comment by rwkay** [comment]
I am not sure if we can do anything with the browser side as that is handled by the browser - unless someone knows of any documentation on if/how this could be done?

**Comment by L-Clouds** [comment]
#5146

**Comment by toby-yoyo** [comment]
In-game audio will override the ring/silent switch following https://github.com/YoYoGames/GameMaker-Bugs/issues/5146. As this issue is a duplicate of that, I am closing this as 'not planned'.
